docs/docs/intro/configure.md

```diff
@@ -375,24 +375,121 @@ marathon {
 </TabItem>
 </Tabs>
 
-### Strict mode
-By default, if one of the test retries succeeds then the test is considered successfully executed. If you require success status only when
-all retries were executed successfully you can enable the strict mode. This may be useful to verify that flakiness of tests was fixed for
-example.
+### Execution strategy
+When executing tests with retries there are multiple trade-offs to be made. Two execution strategies are supported: any success or all success.
+By default, any success strategy is used with fast execution i.e. if one of the test retries succeeds then the test is considered successfully
+executed and all non-started retries are removed.
+
+#### Any success
+Test passes if any of executions is passing. This mode works only if there is no complex sharding strategy applied. This is the default.
+
+<Tabs>
+<TabItem value="YAML" label="Marathonfile">
+
+```yaml
+executionStrategy:
+  mode: ANY_SUCCESS
+```
+
+</TabItem>
+<TabItem value="kts" label="Kotlin DSL">
+
+```kotlin
+marathon {
+  executionStrategy = ExecutionStrategyConfiguration(ExecutionMode.ANY_SUCCESS)
+}
+```
+
+</TabItem>
+<TabItem value="groovy" label="Groovy DSL">
+
+```groovy
+marathon {
+  executionStrategy = ExecutionStrategyConfiguration(ExecutionMode.ANY_SUCCESS)
+}
+```
+
+</TabItem>
+</Tabs>
+
+:::info
+
+Complex sharding with any success mode doesn't make sense when user asks for N number of tests to run explicitly, and we pass on the first one.
+
+:::
+
+#### All success
+Test passes if and only if all the executions are passing. This mode works only if there are no retries, i.e. no complex flakiness strategy, no retry strategy.
+
+<Tabs>
+<TabItem value="YAML" label="Marathonfile">
+
+```yaml
+executionStrategy:
+  mode: ALL_SUCCESS
+```
+
+</TabItem>
+<TabItem value="kts" label="Kotlin DSL">
+
+```kotlin
+marathon {
+  executionStrategy = ExecutionStrategyConfiguration(ExecutionMode.ALL_SUCCESS)
+}
+```
+
+</TabItem>
+<TabItem value="groovy" label="Groovy DSL">
+
+```groovy
+marathon {
+  executionStrategy = ExecutionStrategyConfiguration(ExecutionMode.ALL_SUCCESS)
+}
+```
+
+</TabItem>
+</Tabs>
+
+:::info
+
+Adding retries with retry/flakiness strategies means users wants to trade-off cost for reliability, i.e. add more retries and pass if one 
```

Comment:
```suggestion
Adding retries with retry/flakiness strategies means users want to trade off cost for reliability, i.e. add more retries and pass if one 
```

Reply:
Cost is how much one pays with $ for the test runs

Reply:
done